### DISCUSSION TITLE
What happens to auth.sessions / refresh_tokens on the server?

### DISCUSSION THREAD

**Discussion by hyperknot** [supabase/supabase#12133] [Questions]
I'm trying to understand persistence Supabase Auth in this discussion: https://github.com/supabase/supabase/discussions/11100

So far it's quite lonely, I'm only answering myself. One question I cannot find an answer to is what happens with the auth.sessions / refresh_tokens on the server?

- If these are cleaned up at regular intervals, when exactly and how can we control this? This would be pretty much the most important setting on the dashboard, much more important than the "JWT expiry limit".

- If these are not cleaned up then how do people use Supabase in production? I mean these tables would grow indefinitely with mostly useless leftover data. Over time, performance will degrade and you'll pay for this in storage costs.

Can someone from Supabase comment on this? I've spent a few hours debugging the client behaviour on my own, it'd be so nice if someone could shed some light on the server part.

**Comment by hf** [comment]
Hey @hyperknot,

We have a PR open to deal with this supabase/gotrue#875. It should be going in fairly soon.

We use GoTrue for app.supabase.com and aside from having somewhat large storage we do not experience any performance issues because of the large tables. In general you shouldn't worry about an implementation detail like this for your app.

I'll transfer this issue to the main Supabase repo and convert it as a discussion.

**Comment by jwhits** [comment]
@hf I too had the question about cleanup of the session and token tables. I can see there is now the following SQL to cleanup the session table

`fmt.Sprintf("delete from %q where id in (select id from %q where not_after < now() - interval '72 hours' limit 10 for update skip locked);", tableSessions, tableSessions)`

However the column not_after is allowed nulls and I can't see in the code where it gets a value from so the cleanup would never happen?
